CMakeLists.txt

```diff
@@ -16,11 +16,35 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # --------------------------------------------------------------------------- #
 # Find gz-sim and dependencies.
-find_package(gz-cmake3 REQUIRED)
-set(GZ_CMAKE_VER ${gz-cmake3_VERSION_MAJOR})
 
-gz_find_package(gz-sim7 REQUIRED)
-set(GZ_SIM_VER ${gz-sim7_VERSION_MAJOR})
+# Harmonic
+if("$ENV{GZ_VERSION}" STREQUAL "harmonic")
+  find_package(gz-cmake3 REQUIRED)
+  set(GZ_CMAKE_VER ${gz-cmake3_VERSION_MAJOR})
+
+  gz_find_package(gz-sim8 REQUIRED)
+  set(GZ_SIM_VER ${gz-sim8_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Harmonic")
+# Garden
+elseif("$ENV{GZ_VERSION}" STREQUAL "garden")
+  find_package(gz-cmake3 REQUIRED)
+  set(GZ_CMAKE_VER ${gz-cmake3_VERSION_MAJOR})
+
+  gz_find_package(gz-sim7 REQUIRED)
+  set(GZ_SIM_VER ${gz-sim7_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Garden")
+# Default to Garden
+else()
```

Comment:
You could reduce code duplication with setting a default for a temporary variable.
https://cmake.org/cmake/help/latest/variable/ENV.html

This would also protect against setting an invalid (mistyped) gazebo version that is not supported by failing the build.


Something like:
```
if(DEFINED ENV{GZ_VERSION})
    set(gz_version ENV{GZ_VERSION})
else() 
    set(gz_version garden)
endif()

# Harmonic
if("${gz_version}" STREQUAL "harmonic")
  find_package(gz-cmake3 REQUIRED)
  set(GZ_CMAKE_VER ${gz-cmake3_VERSION_MAJOR})

  gz_find_package(gz-sim8 REQUIRED)
  set(GZ_SIM_VER ${gz-sim8_VERSION_MAJOR})

  message(STATUS "Compiling against Gazebo Harmonic")
# Garden
elseif("${gz_version}" STREQUAL "garden")
  find_package(gz-cmake3 REQUIRED)
  set(GZ_CMAKE_VER ${gz-cmake3_VERSION_MAJOR})

  gz_find_package(gz-sim7 REQUIRED)
  set(GZ_SIM_VER ${gz-sim7_VERSION_MAJOR})

  message(STATUS "Compiling against Gazebo Garden")

else()
    message(FATAL_ERROR "Unsupported GZ_VERSION of ${GZ_VERSION}" )
endif()
```

Reply:
Update in 6e8449b, opted not to introduce an additional variable but check for garden or empty env.

Reply:
Looks much better!